### ISSUE TITLE
Review (lab2) by Omar Ormachea (s304811)

### ISSUE THREAD

**Issue by oormacheah** [Marco-Masera/comp_int_2022#4]
Hi,

Overall I have to say that your code is well organized and readable, plus the comments and README file are very helpful for reviewing purposes. It is very well appreciated! 

Getting deeper into the code, I have a couple things I'd like to point out. They are mostly about the notations and theoretical concepts used. There is nothing substantially "wrong" or improvable about your code that I could find.

## Observations
- The representation you chose for the problem definitely constrains the "kind" of GA you can use. In your case, the `elementCovers` array makes it necessary to work with only feasible solutions since the beginning. I would consider this solution to be more of a *Hill Climbing* algorithm with Genetic tweaks.
- I think the terms `POPULATION_SIZE` and `TOURNAMENT_SIZE` are used inaccurately (not that this compromises the quality of your solution, it's just an observation). As far as I understood:
  - `POPULATION_SIZE` should be the number of individuals at the start of each new generation (in your solution, each new generation is consisting only of the offspring). 
  - As for the `TOURNAMENT_SIZE`, it should represent the number of randomly chosen individuals for a "tournament" between them (e.g. for parent selection). In your solution, the parent selection is completely random so there are no "tournaments" happening.

**Comment by Marco-Masera** [comment]
Hi. Thanks for your review.
Regarding the misuse of the terminology you are completely right; that's
because when I started writing the code I had in mind a different
implementation, and as I changed it i did not rename the variables, which I
should have done.
Especially the tournament number became simply the number of individual
chosen for reproduction at each generation.

I'm not sure that restricting the possible genome space to feasible
solutions makes it a Hill Climbing tho, individuals are still generated
mixing the genomes of two others, with no overimposed "direction" of
evolution.


Il Dom 13 Nov 2022, 22:26 Omar Ormachea Hermoza ***@***.***>
ha scritto:

**Comment by oormacheah** [comment]
You are right, I got confused about the terminologies. Starting with a feasible solution is definitely a particular case of GA, but in principle it should be a GA.
